src/main/java/com/mineinabyss/enhancedwater/WaterListener.java

```diff
@@ -13,10 +14,21 @@
 import java.util.UUID;
 
 public class WaterListener implements Listener {
+    private EnhancedWater plugin;
+    private FileConfiguration config;
 
-    double checkRange = 0.3; //multiplier for blocks around player to be checked
-    double fallRange = 1.2; //block distance to check for player to be pushed while falling in water
-    double pushVelocity = 0.2; //velocity to push players off horizontally in waterfall
+    private double checkRange; //multiplier for blocks around player to be checked
+    private double fallRange; //block distance to check for player to be pushed while falling in water
+    private double pushVelocity; //velocity to push players off horizontally in waterfall
+
+    public WaterListener(EnhancedWater plugin) {
+        this.plugin = plugin;
+        this.config = plugin.getConfig();
+
+        checkRange = config.getDouble("checkRange");
```

Comment:
Nitpicky, but does this barf if the config is missing? Also if you saveDefaultConfig each time will it overwrite changes?

Reply:
@Derongan [The plugin regenerates the config from source if it is not present, if you mean the file itself gets missing](https://hub.spigotmc.org/javadocs/spigot/org/bukkit/plugin/Plugin.html#saveDefaultConfig--). This is fine except if we are going to implement a reload command.

But I don't deny the possibillity of an exception being thrown if the config node is missing. We can add some fallback value to the field.